stdlib/universe/join.go

```diff
@@ -604,13 +609,18 @@ func (c *MergeJoinCache) ForEach(f func(flux.GroupKey)) {
 		}
 		f(key)
 	})
+	return err
 }
 
 // ForEachWithContext iterates over each table in the output stream
-func (c *MergeJoinCache) ForEachWithContext(f func(flux.GroupKey, execute.Trigger, execute.TableContext)) {
+func (c *MergeJoinCache) ForEachWithContext(f func(flux.GroupKey, execute.Trigger, execute.TableContext)) error {
 	trigger := execute.NewTriggerFromSpec(c.triggerSpec)
 
+	var err error
 	c.postJoinKeys.Range(func(key flux.GroupKey, value interface{}) {
+		if err != nil {
+			return
+		}
```

Comment:
You could also avoid having to do this error check if this lamdba could return an error, since `Range` could do the check and terminate iteration.